keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -0,0 +1,1308 @@
+
+No
+
+### Scalability
+
+        For alpha, this section is encouraged: reviewers should consider these questions and attempt to answer them.
+
+        For beta, this section is required: reviewers must answer these questions.
+
+        For GA, this section is required: approvers should be able to confirm the previous answers based on experience in the field.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+DeclarativeValidation: Yes, when enabled the feature may impact validation time. Benchmarks will be taken to measure impact.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+DeclarativeValidation: Yes, when enabled the feature may impact validation CPU and RAM usage. Benchmarks will be taken to measure impact.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+        This section must be completed when targeting beta to a release.
+
+        For GA, this section is required: approvers should be able to confirm the previous answers based on experience in the field.
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No change in behavior.
+
+###### What are other known failure modes?
+
+*   Since validation is used to persist objects to storage, if there was a mistake and it permits denies objects that should be persisted a cluster might end up in an unusable state.  As mentioned above we have test mitigations present to prevent this s.t. the validations are identical
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+### Use CEL and OpenAPI libraries directly for K8s Native Types ([KEP-4153](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4153-declarative-validation))
+
+## Infrastructure Needed (Optional)
+
+## Future Work
+### "New K8s Native APIs" Design Partner For Declarative Validation in 1.34   
+For 1.34 the Declarative Validation working group is looking for a design partner to collaborate with aid in the UX, "validator" creation, and developer process of using Declarative Validation. The partnership would be focused around the new k8s native API creation context of declarative validation (vs the migrating of current k8s native APIs).
```

Comment:
Spoke offline w/ @thockin  and @jpbetz regarding the idea of having a design partner for Declarative Validation specifically circle the 1.34 release timeline around the users story of "new k8s API creation".  

@pohly would DRA be interested in pursuing being a design partner for Declarative Validation (re: from discussion in https://github.com/kubernetes/enhancements/pull/5034)?  Happy to discuss more here if so or via slack, etc.  Thanks